### PR TITLE
feat: rule monitor_gke_pod_env_secrets

### LINKED ISSUE #68: [create new rule] gke_pod_env_secrets / no_secrets_in_env_vars
- @BrunoReboul 
  - [x] create an issue
- @MarcFundenberger
- [x] create topic branch named rulename 
- [x] create `.rego` rule file
- [x] create `.yaml` constraint file
- [x] create `readme.md`
- [x] Initiate PR
- @BrunoReboul 
- [x] review PR, iterate
- [x] create system tests, iterate
- [x] merge to dev

### PR DESCRIPTION
**Pull request by MarcFundenberger**
fixes #68 

New rules to check environment variables in GKE pods definitions.

Rule triggers if :
- var name contains keywords and has a value assigned (to avoid violation on variables from k8s secrets, etc.)
- var value matches regex known patterns (even if value is base64 encoded)

Rule also allows var name whitelisting.

### CODE REVIEW

File: services/monitor/instances/monitor_gke_pod_env_secrets/constraints/no_secrets_in_env_vars/constraint.yaml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+kind: GCPGKEPodEnvSecretConstraintV1
+metadata:
+  name: no_secrets_in_env_vars
+  annotations:
+    description: k8s pods with ENV var containing suspicious keywords.
+    category: Secret Management
+spec:
+  severity: critical
+  match:
+    target: ["organization/"]
```
Comment (BrunoReboul) [inline services/monitor/instances/monitor_gke_pod_env_secrets/constraints/no_secrets_in_env_vars/constraint.yaml:25]: Would you remove double quotes?

**Review by BrunoReboul — CHANGES_REQUESTED** [review]

File: services/monitor/instances/monitor_gke_pod_env_secrets/constraints/no_secrets_in_env_vars/constraint.yaml (hunk 1 above)
Comment (MarcFundenberger) [reply services/monitor/instances/monitor_gke_pod_env_secrets/constraints/no_secrets_in_env_vars/constraint.yaml]: done

**Review by BrunoReboul — APPROVED** [review]

File: services/monitor/instances/monitor_gke_pod_env_secrets/monitor_gke_pod_env_secrets.rego [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+deny[{
+	"msg": message,
+	"details": metadata,
+}] {
+	constraint := input.constraint
+	lib.get_constraint_params(constraint, params)
+
+	name_whitelist := lib.get_default(params, "name_whitelist", [])
+	name_contains := lib.get_default(params, "name_contains", [])
+	value_contains := lib.get_default(params, "value_contains", [])
```
Comment (BrunoReboul) [inline services/monitor/instances/monitor_gke_pod_env_secrets/monitor_gke_pod_env_secrets.rego:28]: Same as for gcf:

The check on name contains is case sensitive, so with the constraint set to "PASS" it detects a variable named PASSWORD but it did not detected the variable named password. I know it is not a best practice to define env vars lower case, but as it is possible and may become a way to pass secrets without being detected, I would prefer to chatch them too.

would you make the rule case insensitive ?

**Review by BrunoReboul — CHANGES_REQUESTED** [review]

File: services/monitor/instances/monitor_gke_pod_env_secrets/monitor_gke_pod_env_secrets.rego (hunk 2 above)
Comment (MarcFundenberger) [reply services/monitor/instances/monitor_gke_pod_env_secrets/monitor_gke_pod_env_secrets.rego:28]: Done

**Review by BrunoReboul — APPROVED** [review]

**Pull request merged**